### PR TITLE
Net 7155- Consul API Gateway Controller Stub Work

### PR DESCRIPTION
**Pull request by sarahalsmiller**
### Description
- Stub code for consul API Gateway controller

### Testing & Reproduction steps
- This is basic stub work and will be filled out and tested as part of follow up PRs

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: internal/mesh/internal/controllers/apigateways/controller.go
Comment (jm96441n) [inline internal/mesh/internal/controllers/apigateways/controller.go:33]: :tada: I like this pattern of leaving hints as to where implementation for future tickets go!

**Review by jm96441n — APPROVED** [review]
great work! lgtm, once you resolve the merge conflicts this looks good to go!

**Review by nathancoleman — APPROVED** [review]

**Pull request merged**
